src/modules/folder/folder.usecase.ts

```diff
@@ -810,6 +813,24 @@ export class FolderUseCases {
       folder.id,
       updateData,
     );
+
+    const files = await this.fileUsecases.getFilesByFolderUuid(
+      folder.uuid,
+      FileStatus.TRASHED,
+    );
+    const fileIds = files.map((f) => f.fileId);
+    const updateFileData: Partial<File> = {
+      deleted: false,
+      deletedAt: null,
+      status: FileStatus.EXISTS,
+    };
+
+    await this.fileUsecases.updateManyByFileIdAndUserId(
+      fileIds,
+      user.id,
+      updateFileData,
+    );
```

Comment:
It is better to make a query that updates all the files inside that folder instead of fetching and updating them. We have no limits for files inside a folder, so if an user has 10k files in a folder, those are 10k files you're fetching/looping through here.